Assets/C#/PlayerScripts/PlayerAbilities/Ability_PointSpawner.cs

```diff
@@ -0,0 +1,197 @@
+     */
+
+	/**
+     * Called by the input controller. 
+     */
+	public override void use_UseAbility() {
+		// PlayerAbility puts input for each player around here. 
+		// Since we send another message to get things done, we shouldn't bother with anyone who isn't local
+		// Because we take care of all the networking
+		if (isLocalPlayer || myBase.myInput.isBot()) {
+			// Find the first object colliding in front of us, aim at that if necessary
+
+			Ray ray = Camera.main.ScreenPointToRay (Input.mousePosition);
+
+			Vector3? localPosition;
+			Buf buf;
+
+			RaycastHit[] hits = Physics.SphereCastAll (ray.origin, .5f, ray.direction, spawnRange);
+			foreach (RaycastHit h in hits) {
+				PlayerStats tmpSts;
+				if (tmpSts = h.transform.GetComponentInParent<PlayerStats> ()) {
+					if (tmpSts.gameObject == this.gameObject)
+						continue;
+				}
+				if (h.transform.GetComponent<Collider> ().isTrigger) {
+					continue;
+				}
+				//print ("overriding with object: " + h.transform);
+				localPosition = collided(h);
+				if (localPosition.HasValue) {
+					buf = new Buf ();
+					buf.methodName = OBJECT_SPAWN_METHOD_NAME;
+					buf.vectorList = new Vector3[] { (Vector3)localPosition };
+					this.NotifyAllClientDelegates (buf);
+					return;
+				}
+			}
+
+
+		
+			localPosition = maxRangeReached(ray, spawnRange);
+			if (localPosition.HasValue) {
+				buf = new Buf ();
+				buf.methodName = OBJECT_SPAWN_METHOD_NAME;
+				buf.vectorList = new Vector3[] { (Vector3)localPosition };
+				this.NotifyAllClientDelegates (buf);
+			}
+		}
+
+	}
+		
+
+	/*
+	public void SpawnSpell(PlayerComponent.Buf data, float rangeVaried, Ray ray) {
+		GameObject spawn;
+		RaycastHit hit;
+		bool validHit = false;
+		Physics.Raycast (ray, out hit, 1);
+
+		RaycastHit[] hits = Physics.RaycastAll (ray.origin, ray.direction, spawnRange);
+		foreach (RaycastHit h in hits) {
+			PlayerStats tmpSts;
+			if (tmpSts = h.transform.GetComponentInParent<PlayerStats> ()) {
+				if (tmpSts.gameObject == this.gameObject)
+					continue;
+			}
+			if (h.transform.GetComponent<Collider> ().isTrigger) {
+				continue;
+			}
+			//print ("overriding with object: " + h.transform);
+			hit = h;
+			validHit = true;
+			break;
+		}
+
+		if (validHit) {
+			spawn = collided (hit);
+		}
+		else
+		{
+			spawn = maxRangeReached(ray, spawnRange);
+		}
+
+		OnSpellSpawned (spawn);
+		/*
+		RaycastHit hit;
+		GameObject spawn;
+
+		for (int i = 0; i < 100; i++) {
+			if (Physics.Raycast (ray, out hit, rangeVaried)) {
+
+
+				PlayerStats tmpSts;
+				if (tmpSts = hit.transform.GetComponentInParent<PlayerStats> ()) {
+					if (tmpSts.gameObject == this.gameObject)
+						ray = new Ray (ray.origin + ray.direction * .1f, ray.direction);
+						continue;
+				}
+				spawn = collided (hit);
+
+
+
+			} else {
+				spawn = maxRangeReached (ray, rangeVaried);
+
+			}
+			OnSpellSpawned(spawn);
+			break;
+		}
+
+
+	} */
+
+
+	public virtual Vector3? collided(RaycastHit hit)
```

Comment:
Any reason why these are virtual? They could probably be private too, since I don't think we plan on extending this